src/sentry/issues/auto_source_code_config/code_mapping.py

```diff
@@ -541,3 +571,35 @@ def find_roots(frame_filename: FrameInfo, source_path: str) -> tuple[str, str]:
     # validate_source_url should have ensured the file names match
     # so if we get here something went wrong and there is a bug
     raise UnexpectedPathException("Could not find common root from paths")
+
+
+def get_path_from_module(module: str, abs_path: str) -> tuple[str, str]:
+    """This converts a Java module name and filename into a real path."""
+    if abs_path.find("$") > -1:
+        raise DoesNotFollowJavaPackageNamingConvention
+
+    parts = abs_path.split(".")
+    if len(parts) != 2:  # file name + extension
+        raise DoesNotFollowJavaPackageNamingConvention
+    extension = parts[1]
+
+    parts = module.split(".")
+    if len(parts) <= 1:
+        raise DoesNotFollowJavaPackageNamingConvention
+
+    package_name = None
+    temp_path = None
+    # Find the first uppercase letter after a period to identify class name
+    for i, part in enumerate(parts):
+        if part and part[0].isupper():
```

Comment:
For source context we simply use everything before the last `.` instead of looking for upper case letters. You can also use lowercase letters for classes - not sure how common that is tho. You can also use upper case letters for package names.